### DISCUSSION TITLE
My storage is filling up. Tracking down media storage.

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#22816] [Q&A]
I've ran the 2 commands below and it deleted 10GB's worth of cache.

RAILS_ENV=production /home/mastodon/live/bin/tootctl preview_cards remove --days=1

RAILS_ENV=production /home/mastodon/live/bin/tootctl media remove --days=1

Even after running those 2 commands I'm still running out of space on my instance. My total size of the disk is 50GB and the total amount used is 38GB

When i look in the admin panel it says

Media Storage 23.3 GB

PostgreSQL 945 MB

Redis 38.5 MB

I only have 2 users. My main account and my test user. I have uploaded media 500MB of media but not 23.3GB's worth.

Where is this 23.3GB coming from? How can i track it down.

**Comment by ghost** [comment]
So i was able to track it down. I believe it's caching all this stuff. The users aren't even on my instance.
The media remove command & preview_cards remove command doesn't remove this cache. 

How do i remove it. 


The following directory has 19GB

/home/mastodon/live/public/system/cache/accounts/avatars 
6.5GB

/home/mastodon/live/public/system/cache/accounts/headers
13GB


Can i just hard delete these folders? Since they are just cache I can assume it won't delete anything important?

**Reply by gummipunkt** [reply]
Hi,

just let the magic work:

https://instance.tld/admin/settings/content_retention

set the first and third value to 7 days.

**Reply by sepTN** [reply]
The second one is a little but destructive in nature because it literally deletes people's toots?

**Comment by StrangeMinds1960** [comment]
Not sure if it makes any difference, but are you running those commands as the mastodon user?

Alternatively, you could create an Amazon S3 bucket (or equivalent from elsewhere)

https://github.com/cybrespace/cybrespace-meta/blob/master/s3.md

**Comment by ghost** [comment]
I ended up hard deleting the contents inside both folders "avatars" & "headers" using the linux command line "rm -r folder_name"

/home/mastodon/live/public/system/cache/accounts/avatars/

/home/mastodon/live/public/system/cache/accounts/headers/

It got rid of 19.5GB

WOW. Didn't seem to break anything even after deleting those folders.

I will have to add those commands to the cron

After deleting those folders i checked the storage of my hard drive and it says I am only using a total of 12GB now out of 50GB

WOW. Huge difference. 

So i went back into the admin panel and it still says 
Media storage  22.2 GB 


I used tootctl media usage command still shows 19.5 between the headers and avatars folders. Even though i deleted those folders.

Attachments:    1.11 GB (533 MB local)
Custom emoji:   145 MB (20.8 KB local)
Preview cards:  290 KB
Avatars:        5.19 GB (252 KB local)
Headers:        11.5 GB (1.04 MB local)
Backups:        0 Bytes
Imports:        0 Bytes
Settings:       438 KB


But linux df -H command shows now only 12GB used

Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        52G   12G   41G  23% /



Any way to update the admin panel to show the correct usage?

I forgot to mention I am using (v3.5.3).

**Reply by gummipunkt** [reply]
I think it would be better to try it this way:
https://ricard.dev/improving-mastodons-disk-usage/
